frontend/src/pages/comparisons/ComparisonList.tsx

```diff
@@ -18,19 +18,31 @@ function ComparisonsPage() {
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
   const offset = Number(searchParams.get('offset') || 0);
+  const filteredVideo = String(searchParams.get('video') || '');
   const limit = 20;
 
   function handleOffsetChange(newOffset: number) {
-    history.push(`/comparisons/?offset=${newOffset}`);
+    searchParams.set('offset', newOffset.toString());
+    history.push({ search: searchParams.toString() });
   }
 
   useEffect(() => {
-    UsersService.usersMeComparisonsList({ limit, offset }).then((data) => {
-      setComparisons(data.results);
-      setCount(data.count || 0);
-    });
+    searchParams.get('video')
+      ? UsersService.usersMeComparisonsListFiltered({
+          videoId: filteredVideo,
+          limit: limit,
+          offset: offset,
+        }).then((data) => {
+          setComparisons(data.results);
+          setCount(data.count || 0);
+        })
+      : UsersService.usersMeComparisonsList({ limit, offset }).then((data) => {
+          setComparisons(data.results);
+          setCount(data.count || 0);
+        });
```

Comment:
This code could be slightly refactored to avoid repeating the piece:
```TypeScript
      .then((data) => {
          setComparisons(data.results);
          setCount(data.count || 0);
        });
```

Also we prefer using `await` for asynchronous calls, so we are slowly removing the `.then` when possible (@tournesol-app/maintainers do you confirm?)


```suggestion
    const process = async () => {
      const comparisonsRequest = await (filteredVideo
        ? UsersService.usersMeComparisonsListFiltered({
            videoId: filteredVideo,
            limit,
            offset,
          })
        : UsersService.usersMeComparisonsList({ limit, offset }));
      setComparisons(comparisonsRequest.results);
      setCount(comparisonsRequest.count || 0);
    };
    process();
```